### PR TITLE
Subscription management: Migrate `SubscriptionManager`  to client package

### LINKED ISSUE #74887: Migrate all the calypso client dependent components from subscription-manager package over to the client
## Related conversations
- https://github.com/Automattic/wp-calypso/pull/74733#discussion_r1145377225
- https://github.com/Automattic/wp-calypso/pull/74733#discussion_r1146568724

### PR DESCRIPTION
**Pull request by mashikag**
Fixes https://github.com/Automattic/wp-calypso/issues/74887

## Proposed Changes

* Migrate `SubscriptionManager` to client package because it is directly dependent on components from that package
  * the move to do such migration was motivated by the comment from @noahtallen on another issue (see [the comment](https://github.com/Automattic/wp-calypso/pull/74733#discussion_r1145377225))
* Remove the `subscription-manager` package as there is no need for it anymore

## Testing Instructions

* Go to http://calypso.localhost:3000/subscriptions/settings, ensuring you have a valid `subkey` value in your cookie which is pointing to `calypso.localhost`

## Pre-merge Checklist

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-ajp-p2)?

### CODE REVIEW

File: client/landing/subscriptions/hooks/useSubheaderText.tsx [hunk 1]
```diff
@@ -0,0 +1,27 @@
+import { Reader } from '@automattic/data-stores';
```
Comment (JuanLucha) [inline client/landing/subscriptions/hooks/useSubheaderText.tsx]: This file name is useSubheaderText.tsx. Shouldn't it be use-subheader-text.tsx?

Comment (mashikag) [reply client/landing/subscriptions/hooks/useSubheaderText.tsx]: Well spotted @JuanLucha ! It is difficult for me to get used to the kebab case. I will fix it in a moment. 🙇‍♂️

Comment (JuanLucha) [reply client/landing/subscriptions/hooks/useSubheaderText.tsx]: Me too! The official react docs use the Pascal case themselves, so I guess that makes a lot of us get used to that nomenclature

**Review by JuanLucha — APPROVED** [review]
Great job!

**Pull request merged**